apps/src/templates/instructions/FeedbackStudentReviewState.jsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import i18n from '@cdo/locale';
+import {KeepWorkingBadge} from '@cdo/apps/templates/progress/BubbleBadge';
+import {ReviewStates} from '@cdo/apps/templates/types';
+import color from '@cdo/apps/util/color';
+
+class FeedbackStudentReviewState extends React.Component {
+  static propTypes = {
+    latestReviewState: PropTypes.oneOf(Object.keys(ReviewStates)),
+    isAwaitingTeacherReview: PropTypes.bool
+  };
+
+  renderCompleted() {
+    return (
+      <div style={styles.studentReviewState}>{i18n.reviewedComplete()}</div>
+    );
+  }
+
+  renderKeepWorking() {
+    return (
+      <div style={styles.studentReviewState}>
+        <KeepWorkingBadge style={styles.keepWorkingBadge} />
+        <span style={styles.keepWorking}>{i18n.keepWorking()}</span>
+      </div>
+    );
+  }
+
+  renderAwaitingReview() {
+    return (
+      <div style={styles.studentReviewState}>
+        <KeepWorkingBadge style={styles.keepWorkingBadge} />
+        <span>{i18n.waitingForTeacherReview()}</span>
+      </div>
+    );
+  }
+
+  render() {
+    const {latestReviewState, isAwaitingTeacherReview} = this.props;
+
+    if (latestReviewState === ReviewStates.completed) {
+      return this.renderCompleted();
+    } else if (isAwaitingTeacherReview) {
```

Comment:
Will `latestReviewState === ReviewStates.keepWorking` in this case?  If so, I'm wondering if having this be `latestReviewState === ReviewStates.keepWorking && isAwaitingTeacherReview` would make the intent and relationship between latestReviewState and isAwaitingReview clearer?  If we do that, I would also change 45 to `latestReviewState === ReviewState.keepWorking && !isAwaitingTeacherReview`.  Or possibly combining the two cases and using a ternary.

Reply:
isAwaitingTeacherReview already takes that into account by definition so I'm hesitant to add that check. I'd be open to checking `else if (latestReviewState === ReviewState.keepWorking)` and then return a ternary that depends on isAwaitingTeacherReview

Reply:
Ah interesting.  It's not clear from here that `isAwaitingTeacherReview` already checks `ReviewStates`.  I noticed this because the logic always seems a bit more straightforward to me when the checks in a series of if statements are "parallel".  I'm explaining my reasoning because I'm not sure how best to fix (or leave as-is), up to you.  :-)

Reply:
I have an interesting idea for a refactor here which may clarify this, but I don't want to add to this PR. I've taken some notes and will come back to this.